src/installer/corehost/cli/hostpolicy/hostpolicy_context.cpp

```diff
@@ -114,15 +114,38 @@ int hostpolicy_context_t::initialize(hostpolicy_init_t &hostpolicy_init, const a
     fx_definition_vector_t::iterator fx_end;
     resolver.get_app_fx_definition_range(&fx_begin, &fx_end);
 
+    pal::string_t resolved_frameworks;
     pal::string_t app_context_deps_str;
-    fx_definition_vector_t::iterator fx_curr = fx_begin;
-    while (fx_curr != fx_end)
+    bool is_app = host_mode != host_mode_t::libhost;
+
+    for (const auto& fx : fx_definitions)
     {
-        if (fx_curr != fx_begin)
+        if (!app_context_deps_str.empty())
+        {
             app_context_deps_str += _X(';');
+        }
 
-        app_context_deps_str += (*fx_curr)->get_deps_file();
-        ++fx_curr;
+        app_context_deps_str += fx->get_deps_file();
```

Comment:
Before your change the app.deps.json was not included in this string when running in libhost mode (since only real frameworks should be reported then). Now it will include it.
Can you please adapt some of the existing libhost tests and validate this property and that it doesn't regress with this change?